pkg/controller/storagecluster/reconcile.go

```diff
@@ -173,6 +175,29 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		if phaseErr != nil {
 			reqLogger.Error(phaseErr, "Failed to set PhaseDeleting")
 		}
+
+		// Check if the storagecluster is labelled for data deletion, if yes
+		// label the child resources for data deletion too before issuing delete
+		if v, found := instance.ObjectMeta.Labels[CleanupPolicy]; found {
+			if strings.Compare(v, CleanupPolicyDelete) == 0 {
+				cephCluster := &cephv1.CephCluster{}
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: generateNameForCephCluster(instance), Namespace: instance.Namespace}, cephCluster)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						reqLogger.Info("CephCluster not found")
+					} else {
+						return reconcile.Result{}, fmt.Errorf("Unable to retrive cephCluster: %v", err)
+					}
+				} else {
+					cephCluster.Spec.CleanupPolicy.DeleteDataDirOnHosts = "yes-really-destroy-data"
```

Comment:
```suggestion
					cephCluster.Spec.CleanupPolicy.DeleteDataDirOnHosts = CleanupPolicyDelete
```

Reply:
Please add:

```
cephCluster.Spec.CleanupPolicy.SanitizeDiskSpec.SanitizeDiskConfirmation = CleanupDiskPolicyDelete
```

Reply:
Forget this.

Reply:
To be clear, we still need the following set:
```
cephCluster.Spec.CleanupPolicy.Confirmation = CleanupPolicyDelete
```
The `Confirmation` field is new in rook, so you'll need to update the reference for the latest rook to pick it up.